core/arch/arm/plat-ls/main.c

```diff
@@ -161,3 +164,38 @@ void main_secondary_init_gic(void)
 {
 	gic_cpu_init(&gic_data);
 }
+
+#ifdef CFG_HW_UNQ_KEY_REQUEST
+
+#include <types_ext.h>
+int get_hw_unique_key(uint64_t smc_func_id, uint64_t in_key, uint64_t size);
+
+/*
+ * Issued when requesting to Secure Storage Key for secure storage.
+ *
+ * SiP Service Calls
+ *
+ * Register usage:
+ * r0/x0	SMC Function ID, OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY
+ */
+#define OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY			0xFF14
+#define OPTEE_SMC_FAST_CALL_SIP_LS_HW_UNQ_KEY \
+	OPTEE_SMC_CALL_VAL(OPTEE_SMC_32, OPTEE_SMC_FAST_CALL, \
+			   OPTEE_SMC_OWNER_SIP, \
+			   OPTEE_SMC_FUNCID_SIP_LS_HW_UNQ_KEY)
+
+void tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
+{
+	int ret = 0;
+
+	ret = get_hw_unique_key(OPTEE_SMC_FAST_CALL_SIP_LS_HW_UNQ_KEY,
+			virt_to_phys(&hwkey->data[0]), sizeof(hwkey->data));
+
+	/* Invalidate cache to fetch data from external memory */
+	cache_op_inner(DCACHE_AREA_INVALIDATE,
```

Comment:
What is preventing these cache lines to be written back to external memory just before this call?
The supplied buffer isn't guaranteed to only cover complete cache lines so the stack is likely to be corrupted when this function is called from `tee_fs_init_key_manager()`.
If this platform has some secure memory mapped as uncached it would be much easier to use that as a bounce buffer.

Reply:
Removing this code for invalidating the cache to fetch data from external memory.
Done this change in the code running in EL3 mode.